akka-stream/src/main/scala/akka/stream/scaladsl/Hub.scala

```diff
@@ -134,9 +200,16 @@ private[akka] class MergeHub[T](perProducerBufferSize: Int)
         true
       case Deregister(id) =>
         demands.remove(id)
+        if (draining) tryCompleteOnDraining()
```

Comment:
`if (drainingEnabled && draining)` allows us to avoid touching the volatile when not enabled